payment/datastore.go

```diff
@@ -300,19 +302,17 @@ func (pg *Postgres) GetOrderCreds(orderID uuid.UUID) (*[]OrderCreds, error) {
 	return nil, nil
 }
 
-// GetOrderCredsByItemID returns the order credentials for a OrderID
-func (pg *Postgres) GetOrderCredsByItemID(orderID uuid.UUID) (*[]OrderCreds, error) {
-	orderCreds := []OrderCreds{}
-	err := pg.DB.Select(&orderCreds, "select * from order_creds where order_id = $1 and signed_creds is not null", orderID)
-	if err != nil {
+// GetOrderCredsByItemID returns the order credentials for a OrderID by the itemID
+func (pg *Postgres) GetOrderCredsByItemID(orderID uuid.UUID, itemID uuid.UUID) (*OrderCreds, error) {
+	orderCreds := OrderCreds{}
+	err := pg.DB.Get(&orderCreds, "select * from order_creds where order_id = $1 and id = $2 and signed_creds is not null", orderID, itemID)
+	if err == sql.ErrNoRows {
+		return nil, nil
+	} else if err != nil {
 		return nil, err
 	}
 
-	if len(orderCreds) > 0 {
-		return &orderCreds, nil
-	}
-
-	return nil, nil
+	return &orderCreds, nil
```

Comment:
Since you instantiated "orderCreds" this will not be nil, it will be the address of orderCreds I don't think.

Reply:
Not sure I understand this comment 🤔 What are you asking?

Reply:
NVM, for some reason I thought that you wanted to return nil here if you had no results, but realize that you are checking for ErrNoRows which Get will provide... disregard this.